File: src/msa/web-service/portalapi/src/main/java/vn/com/kns/portalapi/application/service/shop/shopOrderDetail/dto/ShopOrderDetailDto.java
package vn.com.kns.portalapi.application.service.shop.shopOrderDetail.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.kns.portalapi.application.service.supply.dto.SupplyDto;
import vn.com.kns.portalapi.core.model.BaseEntityDto;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ShopOrderDetailDto extends BaseEntityDto {
    private Integer quantity;
    private Long price;
    private String unit;

    private Long shopOrderId;
    private SupplyDto supply;
    private Long supplyId;
}

File: src/msa/main-service/authportal-service/src/app/shared/services/shop/shop-order-detail/dto/shop-order-detail-dto.ts
import {ShopOrderDto} from '../../shop-order/dto/shop-order-dto';
import {SupplyDto} from '../../../category/supply/dto/supply-dto';

export class ShopOrderDetailDto {
  id: number | null;
  price: number = 1;
  quantity: number = 1;
  unit: string;

  shopOrderId: number;
  shopOrder: ShopOrderDto = new ShopOrderDto();

  supplyId: number;
  supply: SupplyDto = new SupplyDto();

  // constructor(id?: number | null, supplyId?: number | null, price?: number | null) {
  //   this.id = id;
  //   this.supplyId = supplyId;
  //   this.price = price;
  // }
}
